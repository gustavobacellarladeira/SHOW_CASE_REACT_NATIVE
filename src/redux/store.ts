import { applyMiddleware, combineReducers, createStore } from "redux";
import thunk from "redux-thunk";
import { themeReducer } from "./themeReducer";

const store = createStore(
  combineReducers({ themeReducer }),
  applyMiddleware(thunk)
);

export { store };
