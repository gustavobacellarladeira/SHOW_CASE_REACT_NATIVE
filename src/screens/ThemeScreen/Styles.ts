import styled from "styled-components/native";

export const Container = styled.View`
  flex: 1;
  width: 100%;
  background-color: ${(props: { theme: { colors: { primary: any } } }) =>
    props.theme.colors?.primary || "#000"};
`;

export const SectionPrimary = styled.View`
  height: 10%;
  background-color: ${(props: { theme: { colors: { primary: any } } }) =>
    props?.theme.colors?.primary || "#000"};
  align-items: center;
  justify-content: center;
`;
export const TitleSectionPrimary = styled.Text`
  color: ${(props: { theme: { PRIMARY_TEXT_COLOR: any } }) =>
    props.theme?.PRIMARY_TEXT_COLOR || "#fff"};
  font-size: ${(props: { theme: { FONT_SIZE_TITLE: any } }) =>
    props.theme?.FONT_SIZE_TITLE || "20px"};
  text-align: left;
`;

export const SectionSecondary = styled.View`
  background-color: ${(props: { theme: { colors: { secondary: any } } }) =>
    props.theme.colors?.secondary || "#000"};
  flex: 1;
  flex-direction: row;
  padding: 20px;
  justify-content: space-between;
`;
