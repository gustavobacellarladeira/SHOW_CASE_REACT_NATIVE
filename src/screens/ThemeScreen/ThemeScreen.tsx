import { getImage } from "../../services/images/images";
import React, { useState } from "react";
import { Switch } from "react-native";
import { useTheme } from "styled-components";
import { useThemeRedux } from "../../store/theme/hook";
import {
  Container,
  SectionPrimary,
  SectionSecondary,
  TitleSectionPrimary,
} from "./Styles";

export const ThemeScreen = () => {
  const theme = useTheme();
  const { useSwitchTheme } = useThemeRedux();
  const [isEnabled, setIsEnabled] = useState(false);
  const toggleSwitch = () => {
    useSwitchTheme();
    setIsEnabled((previousState) => !previousState);
  };

  return (
    <Container>
      <SectionPrimary>
        <TitleSectionPrimary>Popular Movies</TitleSectionPrimary>
      </SectionPrimary>

      <SectionSecondary>
        <TitleSectionPrimary>Change Theme</TitleSectionPrimary>

        <Switch
          trackColor={{
            false: theme.colors.primary,
            true: theme.colors.primary,
          }}
          thumbColor={isEnabled ? theme.colors.tertiary : theme.colors.primary}
          ios_backgroundColor="#3e3e3e"
          onValueChange={toggleSwitch}
          value={isEnabled}
        />
      </SectionSecondary>
    </Container>
  );
};
