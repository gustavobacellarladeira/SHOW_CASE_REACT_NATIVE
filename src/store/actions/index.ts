export { REMOVE_MOVIE, SET_MOVIE } from "./movies";
