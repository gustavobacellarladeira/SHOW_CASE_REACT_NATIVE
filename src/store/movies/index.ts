export { moviePopularReducer } from "./popular/reducer";
export { movieTrendingReducer } from "./trendings/reducer";
