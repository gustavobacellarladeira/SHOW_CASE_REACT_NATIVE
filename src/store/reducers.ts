import { combineReducers } from "redux";
import { themeReducer } from "./theme/reducer";
import { movieReducer } from "./movies/index";

export const reducer = combineReducers({
  movies: movieReducer,
  theme: themeReducer,
});
