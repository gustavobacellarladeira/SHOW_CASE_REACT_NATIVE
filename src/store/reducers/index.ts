import { combineReducers } from "redux";
import { movieReducer } from "./movies";

export const reducer = combineReducers({
  movies: movieReducer,
});
