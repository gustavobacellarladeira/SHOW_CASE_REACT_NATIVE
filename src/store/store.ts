import { applyMiddleware, createStore } from "redux";
import { reducer } from "./reducers";
import promiseMiddleware from "redux-promise-middleware";
import promise from "redux-promise-middleware";
import thunk from "redux-thunk";

applyMiddleware(promiseMiddleware)(createStore);

const store = createStore(reducer, {}, applyMiddleware(promise));
export { store };
