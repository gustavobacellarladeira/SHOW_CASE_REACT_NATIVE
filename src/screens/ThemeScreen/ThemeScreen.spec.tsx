import { store } from "../../store/store";
import React from "react";
import { Provider } from "react-redux";
import { ThemeScreen } from "./ThemeScreen";
import { render, fireEvent } from "@testing-library/react-native";
import { ThemeProvider, DefaultTheme } from "styled-components";

const MOCK_THEME: DefaultTheme = {
  mode: "dark",
  PRIMARY_BACKGROUND_COLOR: "#333333",
  PRIMARY_TEXT_COLOR: "#ffffff",
  FONT_SIZE_TITLE: "20px",
  fonts: {
    color: "#ffffff",
    size: "20px",
  },
  colors: {
    primary: "#222831",
    secondary: "#393E46",
    tertiary: "#D65A31",
  },

  tabBottomBar: {
    backgroundColor: "#333",
    focused: "#D65A31",
    inactive: "#ffffff",
  },
};

const rendererWithRedux = (component: any) => {
  return render(
    <Provider store={store}>
      <ThemeProvider theme={MOCK_THEME}>{component}</ThemeProvider>
    </Provider>
  );
};

describe("ThemeScreen testing...", () => {
  it("renders correctly", () => {
    const tree = rendererWithRedux(<ThemeScreen />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("isEnabled correctly", () => {
    const { getByTestId } = rendererWithRedux(<ThemeScreen />);
    const isEnabledComp = getByTestId("isEnabled");
    expect(isEnabledComp).toBeTruthy();
  });

  it("Switch has to change isEnable", () => {
    const { getByTestId } = rendererWithRedux(<ThemeScreen />);
    const switchComp = getByTestId("switchTheme");
    console.log("switchComp VALUE-->", switchComp.props.value);
    // testar o switch ativando a função da props dele switchComp.props.onChange e verificar se o valor do isEnabledComp mudou
    fireEvent.press(switchComp);
    const isEnabledComp = getByTestId("isEnabled");
    expect(isEnabledComp).toBeTruthy();
  });
});
