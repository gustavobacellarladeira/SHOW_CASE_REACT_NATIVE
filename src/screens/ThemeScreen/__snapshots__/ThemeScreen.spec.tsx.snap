// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeScreen testing... renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#000",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#000",
          "height": "10%",
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 20,
            "textAlign": "left",
          },
        ]
      }
    >
      Popular Movies
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#000",
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "space-between",
          "paddingBottom": 20,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 20,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 20,
            "textAlign": "left",
          },
        ]
      }
    >
      Change Theme
    </Text>
    <Text
      testID="isEnabled"
    />
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onTintColor="#222831"
      style={
        Array [
          Object {
            "height": 31,
            "width": 51,
          },
          Object {
            "backgroundColor": "#3e3e3e",
            "borderRadius": 16,
          },
        ]
      }
      testID="switchTheme"
      thumbTintColor="#222831"
      tintColor="#222831"
      value={false}
    />
  </View>
</View>
`;
